src/components/v2/V2Project/V2ProjectSettingsPage/V2ProjectSettings.tsx

```diff
@@ -35,19 +48,99 @@ const items: MenuItem[] = [
 ]
 
 const V2ProjectSettings = () => {
+  const [projectForm] = useForm<ProjectDetailsFormFields>()
+
+  const [loadingSaveChanges, setLoadingSaveChanges] = useState<boolean>()
+  const { projectMetadata, projectId } = useContext(V2ProjectContext)
+
+  const EditV2ProjectDetailsTx = useEditV2ProjectDetailsTx()
+
+  async function onProjectFormSaved() {
```

Comment:
**question** not sure if this will change up the stack, but I think we should isolate this logic to another component. Like `ProjectSettingsPanel.tsx` or somethin